plasmapy/formulary/parameters.py

```diff
@@ -333,9 +369,16 @@ def ion_sound_speed(T_e,
 
     T_i = T_i.to(u.K, equivalencies=u.temperature_energy())
     T_e = T_e.to(u.K, equivalencies=u.temperature_energy())
+    
+    # Assume non-dispersive limit if values for n_e (or k) are not specified
+    if n_e is None or k is None:
+        klD2 = 0.0
+    else:
+        lambda_D = Debye_length(T_e, n_e)
+        klD2 = (k * lambda_D) ** 2
```

Comment:
As the [above comment suggests](https://github.com/PlasmaPy/PlasmaPy/pull/700/files#r336746787), place warning in this `if-else` statement

```python
if n_e is None or k is None:
    klD2 = 0.0

    if n_e is None and k is None:
        warnings.warn("say stuff")
else:
    lambda_D = Debye_length(T_e, n_e)
    klD2 = (k * lambda_D) ** 2
```

Reply:
Do you think the warning should just go within the first if statement (rather than being within the nested if statement)? That way, the function would warn the user whenever the non-dispersive limit is assumed.

Reply:
Or... do you think it would make sense to only call the warning when one of `k` or `n_e` is given, so scripts that are dependant on PlasmaPy and were previously happy to use the non-dispersive limit don't cause warnings to be shown?

Reply:
Good point.  I think the `n_e=None` and `k=None` case should not throw a warning since it is the default state.  @StanczakDominik correct me if you think otherwise.  In that case, I would restructure it like...

```python
klD2 = 0.0
if (n_e is None) ^ (k is None):
    warnings.warn("say stuff")
elif n_e is not None and k is not None:
    lambda_D = Deby_length(T_e, n_e)
    klD2 = (k * lambda_D) ** 2
```

The parenthesis are needed in this case since the xor operator `^` only acts on booleans.  The parenthesis force the conditionals into booleans before the xor operation is poerformed.

Reply:
Neat! I never knew there was an XOR operator in python. I see what you’re saying about the default case. That makes sense.

Reply:
Yeah, defaults should probably not throw warnings! I haven't (consciously) seen XOR in Python yet either. I like the proposed way of doing things :)